src/main/java/Ui.java

```diff
@@ -0,0 +1,147 @@
+import java.time.LocalDate;
+import java.util.Scanner;
+import java.util.ArrayList;
+
+/**
+ * Ui allows interactions with the user and prompts user for command
+ */
+public class Ui {
+    private final Scanner sc;
+    private boolean exit;
+
+    public Ui() {
+        this.sc = new Scanner(System.in);
+        this.exit = false;
+    }
+
+    public void greet() {
+        String logo =  " ____         _        \n"
+                    + "|  _ \\ _   _| | _____ \n"
+                    + "| | | | | | | |/ / _ \\\n"
+                    + "| |_| | |_| |   <  __/\n"
+                    + "|____/ \\__,_|_|\\_\\___|\n";
+        String greeting = "Hello from\n" + logo;
+        greeting += "    ____________________________________________________________\n";
+        greeting += "     Hello! I'm Duke\n";
+        greeting += "     What can I do for you?\n";
+        greeting += "    ____________________________________________________________\n";
+        System.out.println(greeting);
+    }
+
+    public TaskList process(TaskList tasks) {
+        String input = this.next();
+        Parser parser = new Parser();
+        parser = parser.parse(input);
+        if (parser.getTaskType().equals("bye")) {
+            String msg = this.format("     Bye. Hope to see you again soon!");
+            System.out.println(msg);
+            this.exit = true;
+            return tasks;
+        } else if (parser.getTaskType().equals("list")) {
+            String msg = "     Here are the tasks in your list:\n";
+            ArrayList<Task> taskList = tasks.getList();
+            int count = 1;
+            for (Task t : taskList) {
+                if (count == taskList.size()) {
+                    msg += "     " + count + "." + t.toString();
```

Comment:
There is a lot of the same indentation, maybe it can be extracted into a variable, which can be changed easily later？